src/interactive/workspace.ts

```diff
@@ -1,45 +1,77 @@
 import * as vscode from 'vscode';
+import * as repl from './repl';
 
 let g_terminal: vscode.Terminal = null
-let g_replVariables: string = '';
 
-export class REPLTreeDataProvider implements vscode.TreeDataProvider<string> {
-    private _onDidChangeTreeData: vscode.EventEmitter<string | undefined> = new vscode.EventEmitter<string | undefined>();
-    readonly onDidChangeTreeData: vscode.Event<string | undefined> = this._onDidChangeTreeData.event;
+interface WorkspaceVariable {
+    name: string,
+    type: string,
+    value: string
+}
+
+let g_replVariables: WorkspaceVariable[] = undefined;
+
+export class REPLTreeDataProvider implements vscode.TreeDataProvider<WorkspaceVariable> {
+    private _onDidChangeTreeData: vscode.EventEmitter<WorkspaceVariable | undefined> = new vscode.EventEmitter<WorkspaceVariable | undefined>();
+    readonly onDidChangeTreeData: vscode.Event<WorkspaceVariable | undefined> = this._onDidChangeTreeData.event;
 
     refresh(): void {
         this._onDidChangeTreeData.fire(undefined);
     }
 
-    getChildren(node?: string) {
+    getChildren(node?: WorkspaceVariable) {
         if (node) {
-            return [node]
+            return []
         }
         else {
-            if (g_terminal) {
-                return g_replVariables.split(';').slice(1)
+            if (g_replVariables) {
+                return g_replVariables
             }
             else {
-                return ['no repl attached']
+                return []
             }
         }
     }
 
-    getTreeItem(node: string): vscode.TreeItem {
-        let treeItem: vscode.TreeItem = new vscode.TreeItem(node)
+    getTreeItem(node: WorkspaceVariable): vscode.TreeItem {
+        let treeItem = new vscode.TreeItem(`${node.name}:`)
+        treeItem.description = node.value;
+        treeItem.tooltip = node.type;
+        treeItem.contextValue = 'globalvariable';
```

Comment:
Can we make this depend on whether it's actually possible for us to display them in VSCode? That way the button would only show up for tables and plots and stuff?

Reply:
Not really, the problem are tables: in the Queryverse world you can have a query where you can't tell whether something is a table or not until you try to iterate the rows for it, i.e. until you actually try to materialize the results of the query. So that pretty fundamentally prevents us from telling with certainty whether we should show the button or not.

We could, though, narrow things down a lot: the case I described above is rare, and in very many cases we should actually be able to tell whether we can show the value or not.

I've put this onto my list for another follow up PR, ok?